glib/src/subclass/type_module.rs

```diff
@@ -0,0 +1,145 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{subclass::prelude::*, translate::*, Cast, IsA, TypeModule};
+
+pub trait TypeModuleObserver<T: IsA<TypeModule>> {
+    fn on_load(type_module: &T) -> bool;
+    fn on_unload(type_module: &T);
+}
+
+pub trait TypeModuleImpl: ObjectImpl + TypeModuleImplExt {
+    // rustdoc-stripper-ignore-next
+    /// Loads the module, registers one or more types using
+    /// [`register_module_type`] and registers one or more interfaces using
+    /// [`register_module_interface`] (see [`TypeModule`]).
+    ///
+    /// [`register_module_type`]: ../types/fn.register_module_type.html
+    /// [`register_module_interface`]: ../interface/fn.register_module_interface.html
+    /// [`TypeModule`]: ../../gobject/auto/type_module/struct.TypeModule.html
+    fn load(&self) -> bool;
+    // rustdoc-stripper-ignore-next
+    /// Unloads the module (see [`TypeModuleExt::unuse`]).
+    ///
+    /// [`TypeModuleExt::unuse`]: ../../gobject/auto/type_module/trait.TypeModuleExt.html#method.unuse
+    fn unload(&self);
+}
+
+pub trait TypeModuleImplExt: ObjectSubclass {
+    fn parent_load(&self) -> bool;
+    fn parent_unload(&self);
+}
+
+impl<T: TypeModuleImpl> TypeModuleImplExt for T {
+    fn parent_load(&self) -> bool {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().parent_class() as *const gobject_ffi::GTypeModuleClass;
+
+            let f = (*parent_class)
+                .load
+                .expect("No parent class implementation for \"load\"");
```

Comment:
These are optional or not? So if they don't exist you'd simply do nothing instead of panicking

Reply:
I checked the C code and it indeed requires implementations of these. In that case, panicking here is correct.